pkg/mimir/mimir.go

```diff
@@ -756,10 +760,19 @@ func (t *Mimir) Run() error {
 		return err
 	}
 
+	// `ready` is used to allow our signal handler to immediately mark this component
+	// as "not ready" before starting the process of shutting down all dependent services
+	ready := atomic.NewBool(true)
```

Comment:
"ready" => "shutdownRequested". Renaming `ready` to `shutdownRequested` will toggle the checks, but documents the purpose better.

We can then also simplify the comment, eg: `// Used to delay shutdown, but return "not ready" during this delay`.

Reply:
I agree with Peter's feedback.

Reply:
Fixed!